### PR TITLE
chore: Implement TypeScript for Ramp folder

### LINKED ISSUE #10204: Chore: Typescript type coverage UI Folder part 3
### What is this about?

The following strategy to get this done without causing friction to others work will be:

Start to uncomment, commented lines at tsconfig.lint.json
Fix the flagged typescript issues by yarn lint:tsc of the following chunk of files
Revert the uncommented lines at tsconfig.lint.json
Open a PR against the issue

  **UI Folder part 3**
   ```  
     1  app/components/UI/Ramp/components/Account.tsx:61
     2  app/components/UI/Ramp/components/CurrencyChevron.tsx:30
     2  app/components/UI/Ramp/components/ErrorView.tsx:88
     2  app/components/UI/Ramp/components/InfoAlert.tsx:142
     2  app/components/UI/Ramp/components/Keypad.tsx:69
     4  app/components/UI/Ramp/components/modals/FiatSelectModal.tsx:168
     2  app/components/UI/Ramp/components/OrderDetails.tsx:103
     2  app/components/UI/Ramp/components/PaymentMethod.tsx:174
    16  app/components/UI/Ramp/components/PaymentMethodIcon.tsx:89
     2  app/components/UI/Ramp/components/PaymentMethodSelector.tsx:54
     2  app/components/UI/Ramp/components/Quote/Quote.tsx:154
     2  app/components/UI/Ramp/components/RegionAlert.tsx:77
     6  app/components/UI/Ramp/components/RegionModal.tsx:357
     4  app/components/UI/Ramp/components/TokenSelectModal.tsx:201
     3  app/components/UI/Ramp/hooks/useBalance.ts:46
     1  app/components/UI/Ramp/hooks/useGasPriceEstimation.test.ts:74
     1  app/components/UI/Ramp/hooks/useHandleSuccessfulOrder.ts:48
     1  app/components/UI/Ramp/sdk/index.tsx:261
     8  app/components/UI/Ramp/Views/BuildQuote/BuildQuote.tsx:245
```

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by smilingkylan**
## **Description**

Purpose of this task is to remove many of the Typescript errors from within the `app/components/UI/Ramp` folder.

## **Related issues**

Fixes: https://github.com/orgs/MetaMask/projects/60/views/6?pane=issue&itemId=69416218

## **Manual testing steps**

1. Go to `tsconfig.lint.json`
2. Uncomment the `// "app/components/**/*",` line
3. `yarn lint:tsc`
4. See if any errors show up from the `app/component/UI/Ramp` folder

### **Before**

     2  app/components/UI/Ramp/hooks/useBalance.ts:57
     1  app/components/UI/Ramp/hooks/useGasPriceEstimation.test.ts:73
     1  app/components/UI/Ramp/hooks/useHandleSuccessfulOrder.ts:48
     1  app/components/UI/Ramp/orderProcessor/aggregator.test.ts:260
     2  app/components/UI/Ramp/sdk/index.tsx:254
    14  app/components/UI/Ramp/utils/index.test.ts:112
     9  app/components/UI/Ramp/Views/NetworkSwitcher/NetworkSwitcher.test.tsx:18
     1  app/components/UI/Ramp/Views/OrdersList/OrdersList.test.tsx:119
     9  app/components/UI/Ramp/Views/Quotes/Quotes.tsx:146
     2  app/components/UI/Ramp/Views/SendTransaction/SendTransaction.test.tsx:35
     4  app/components/UI/Ramp/Views/SendTransaction/SendTransaction.tsx:114

### **After**

Please note that the remaining issues with the `Quotes.tsx` will be handled in a separate issue

```
     9  app/components/UI/Ramp/Views/Quotes/Quotes.tsx:146
```

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/Ramp/hooks/useHandleSuccessfulOrder.ts [hunk 1]
```diff
@@ -45,7 +49,7 @@ function useHandleSuccessfulOrder() {
       if (
         // TODO: Replace "any" with type
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (tommasini) [inline app/components/UI/Ramp/hooks/useHandleSuccessfulOrder.ts:51]:
```suggestion

```

File: app/components/UI/Ramp/sdk/index.tsx [hunk 2]
```diff
@@ -251,6 +251,7 @@ export const RampSDKProvider = ({
       setRampType,
 
       intent,
+      // @ts-expect-error - TODO: need help from Ramps team to fix this
```
Comment (tommasini) [inline app/components/UI/Ramp/sdk/index.tsx:254]: Lets replace this by the "// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck - Ramps team ownership"

File: app/components/UI/Ramp/Views/Quotes/Quotes.tsx [hunk 3]
```diff
@@ -1,3 +1,6 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck issue with TS cli finding error but not VS Code plugin
```
Comment (wachunei) [inline app/components/UI/Ramp/Views/Quotes/Quotes.tsx:2]: can you add more details to this? is this only a couple of lines and what lines?

File: app/components/UI/Ramp/Views/Checkout.tsx [hunk 4]
```diff
@@ -1,3 +1,6 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck - Ramps team ownership
```
Comment (wachunei) [inline app/components/UI/Ramp/Views/Checkout.tsx:2]: can you add more details to this check skip? is this only a couple of lines and what lines?

File: app/components/hooks/useAccounts/useAccounts.ts [hunk 5]
```diff
@@ -49,7 +51,9 @@ const useAccounts = ({
   const conversionRate = useSelector(selectConversionRate);
   const currentCurrency = useSelector(selectCurrentCurrency);
   const ticker = useSelector(selectTicker);
-  const internalAccounts = useSelector(selectInternalAccounts);
+  const internalAccounts: InternalAccountTypes[] = useSelector(
+    selectInternalAccounts,
+  );
```
Comment (wachunei) [inline app/components/hooks/useAccounts/useAccounts.ts:56]: I don't believe this is needed, the selector function should be typed and the returned value from `useSelector` must be inferred. 

In this case, is the returned value from the `useSelector` different than `InternalAccountTypes[]` ?

File: app/components/UI/Ramp/Views/Quotes/Quotes.tsx (hunk 3 above)
Comment (kylanhurt) [reply app/components/UI/Ramp/Views/Quotes/Quotes.tsx:2]: For some reason our IDEs are not picking up on the TS errors in this file, but the `yarn lint:tsc` command IS detecting the errors.

```
app/components/UI/Ramp/Views/Quotes/Quotes.tsx:146:32 - error TS2349: This expression is not callable.
  Each member of the union type '{ <S extends QuoteResponse | QuoteError>(predicate: (value: QuoteResponse | QuoteError, index: number, array: (QuoteResponse | QuoteError)[]) => value is S, thisArg?: any): S[]; (predicate: (value: QuoteResponse | QuoteError, index: number, array: (QuoteResponse | QuoteError)[]) => unknown, thisArg?: any): (QuoteRes...' has signatures, but none of those signatures are compatible with each other.

146       const allQuotes = quotes.filter(
                                   ~~~~~~

app/components/UI/Ramp/Views/Quotes/Quotes.tsx:147:10 - error TS7006: Parameter 'quote' implicitly has an 'any' type.

147         (quote): quote is QuoteResponse | SellQuoteResponse => !quote.error,
             ~~~~~

app/components/UI/Ramp/Views/Quotes/Quotes.tsx:149:64 - error TS7031: Binding element 'provider' implicitly has an 'any' type.

149       const highlightedPreviouslyUsed = allQuotes.findIndex(({ provider }) =>
                                                                   ~~~~~~~~

app/components/UI/Ramp/Views/Quotes/Quotes.tsx:246:51 - error TS7031: Binding element 'provider' implicitly has an 'any' type.

246       provider_onramp_list: filteredQuotes.map(({ provider }) => provider.name),
                                                      ~~~~~~~~

app/components/UI/Ramp/Views/Quotes/Quotes.tsx:247:55 - error TS7031: Binding element 'provider' implicitly has an 'any' type.

247       previously_used_count: filteredQuotes.filter(({ provider }) =>
                                                          ~~~~~~~~

app/components/UI/Ramp/Views/Quotes/Quotes.tsx:770:38 - error TS7006: Parameter 'quote' implicitly has an 'any' type.

770               highlightedQuotes.map((quote, index) => (
                                         ~~~~~

app/components/UI/Ramp/Views/Quotes/Quotes.tsx:770:45 - error TS7006: Parameter 'index' implicitly has an 'any' type.

770               highlightedQuotes.map((quote, index) => (
                                                ~~~~~

app/components/UI/Ramp/Views/Quotes/Quotes.tsx:895:37 - error TS7006: Parameter 'quote' implicitly has an 'any' type.

895                 filteredQuotes.map((quote, index) => (
                                        ~~~~~

app/components/UI/Ramp/Views/Quotes/Quotes.tsx:895:44 - error TS7006: Parameter 'index' implicitly has an 'any' type.

895                 filteredQuotes.map((quote, index) => (
```

File: app/components/UI/Ramp/Views/SendTransaction/SendTransaction.tsx [hunk 6]
```diff
@@ -153,6 +156,9 @@ function SendTransaction() {
     try {
       trackEvent(
         'OFFRAMP_SEND_TRANSACTION_INVOKED',
+        //@ts-expect-error - TODO: Ramps team needs to resolve discrepancy between
+        // transactionAnalyticsPayload expecting chain_id_source to be a string
+        // but RampTransaction type / interface expecting it to be a number
```
Comment (wachunei) [inline app/components/UI/Ramp/Views/SendTransaction/SendTransaction.tsx:161]: Great catch, will update it in an upcoming PR.

File: app/components/UI/Ramp/types/analytics.ts [hunk 7]
```diff
@@ -274,7 +274,7 @@ interface RampError {
   currency_destination?: string;
 }
 
-interface RampTransaction {
```
Comment (wachunei) [inline app/components/UI/Ramp/types/analytics.ts:277]: Is this needed? We currently use this as a base interface other interfaces extend.

Comment (kylanhurt) [reply app/components/UI/Ramp/types/analytics.ts:277]: @wachunei What is the issue? I just added an `export` before it. Shouldn't affect much(?)

Comment (kylanhurt) [reply app/components/UI/Ramp/types/analytics.ts]: Nevermind, I reverted the export since it wasn't used

**Review by wachunei — APPROVED** [review]

**Review by tommasini — APPROVED** [review]

**Pull request merged**